File: Make the array beautiful.py
from typing import List

class Solution:
    def makeBeautiful(self, arr: List[int]) -> List[int]:
        # code here
        def check(a, b):
            return (a < 0 and b >= 0) or (a >= 0 and b < 0)
        
        st = []
        for i in range(len(arr)):
            if len(st) > 0 and check(st[-1], arr[i]):
                st.pop()
            else:
                st.append(arr[i])
        
        return st

File: Make the array beautiful.cpp
class Solution {
  public:
    bool check(int a, int b) {
        return ((a < 0 && b >= 0) || (a >= 0 && b < 0));
    }
    vector<int> makeBeautiful(vector<int> arr) {
        vector<int> st;
        for (int i = 0; i < arr.size(); ++i) {
            if (!st.empty() && check(st.back(), arr[i]))
                st.pop_back();
            else
                st.push_back(arr[i]);
        }
        return st;
    }
};
